src/rust/engine/fs/fs_util/src/main.rs

```diff
@@ -268,6 +287,63 @@ fn save_directory(
   )
 }
 
+fn materialize_directory(
+  store: Arc<Store>,
+  destination: &Path,
+  fingerprint: &Fingerprint,
+) -> Result<(), ExitError> {
+  let directory = match store.load_directory_proto(&fingerprint)? {
+    Some(d) => d,
```

Comment:
Seems like a good use case for `unwrap_or_else`

Reply:
`ok_or_else` is my friend :) Definitely nicer than what was there before!